### ISSUE TITLE
[application_controller.erl] `permit_application` call

### ISSUE THREAD

**Issue by tank-bohr** [synrc/rt#3]
https://github.com/erlang/otp/blob/34b56e8f566c6f1aa6023bb7b4096b8f2d709d9e/lib/kernel/src/application_controller.erl#L691-L792

That's not okay either.

**Comment by 5HT** [comment]
Yeah, erlang remoting should be replaced with our own implementation. That requests rewriting of application/supervisor internal protocols as well.
